common/persons_api/api.go

```diff
@@ -0,0 +1,96 @@
+package persons_api
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+)
+
+type Client struct {
+	clientId     string
+	clientSecret string
+	accessToken  string
+	httpClient   *http.Client
+	baseUrl      string
+}
+
+func NewClient(id, secret, baseUrl, authUrl string) (*Client, error) {
+	httpClient := &http.Client{}
+	c := &Client{
+		httpClient:   httpClient,
+		clientId:     id,
+		clientSecret: secret,
+		baseUrl:      baseUrl,
+	}
+	accessToken, err := c.GetAccessToken(authUrl)
+	if err != nil {
+		return nil, err
+	}
+	c.accessToken = accessToken
+	return c, nil
+}
+
+func (c *Client) GetAccessToken(authUrl string) (string, error) {
+	authReqBody, err := json.Marshal(AuthReq{
+		Audience:     "api.sso.mozilla.com",
+		Scope:        "classification:public display:public",
+		GrantType:    "client_credentials",
+		ClientId:     c.clientId,
+		ClientSecret: c.clientSecret})
+	if err != nil {
+		return "", err
+	}
+
+	resp, err := c.httpClient.Post(authUrl, "application/json", bytes.NewBuffer(authReqBody))
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+
+	var authResp AuthResp
+	err = json.Unmarshal(body, &authResp)
+	if err != nil {
+		return "", err
+	}
+
+	return authResp.AccessToken, nil
+}
+
+func (c *Client) GetPersonByEmail(primaryEmail string) (*Person, error) {
+	req, err := http.NewRequest("GET", c.baseUrl+"/v2/user/primary_email/"+primaryEmail, nil)
```

Comment:
One other note here, do we need to ensure `primaryEmail` is URL encoded?

Reply:
No, that's done within `http.NewRequest()`